src/main/java/com/derongan/minecraft/deeperworld/DeeperWorldExtensions.kt

```diff
@@ -0,0 +1,13 @@
+package com.derongan.minecraft.deeperworld
+
+import org.bukkit.entity.Entity
+
+internal tailrec fun Entity.getVehicleRecursive() : Entity? {
+    if(!passengers.any() && vehicle == null){
```

Comment:
I assume the passenger check is to ensure this returns null if the player isn't riding anything? If this is the case, it's still possible for entities to ride the player, which would mess this up, but also I feel like it's easier to just return the player itself when there are no vehicles. You could change the function name to reflect that if you like. The code would basically just be `getVehicleRecursive() ?: return this`.

Reply:
I'd still like to return null if there is no vehicle. I can change the function like this, much nicer than before:
```
internal fun Entity.getVehicleRecursive() : Entity? {
    tailrec fun Entity.internalGetVehicleRecursive() : Entity{
        val currentVehicle = vehicle ?: return this

        return currentVehicle.internalGetVehicleRecursive()
    }

    return vehicle?.internalGetVehicleRecursive()
}
```